pkg/externaldata/externaldata.go

```diff
@@ -16,9 +16,16 @@ import (
 	"flag"
 )
 
-// ExternalDataEnabled indicates if the external data feature is enabled.
-var ExternalDataEnabled *bool
+var (
+	// ExternalDataEnabled indicates if the external data feature is enabled.
+	ExternalDataEnabled *bool
+
+	// ExternalDataClientAuthEnabled indicates if gatekeeper should inject its
+	// TLS certificate into an HTTP client that is used to communicate with providers.
+	ExternalDataClientAuthEnabled *bool
+)
 
 func init() {
 	ExternalDataEnabled = flag.Bool("enable-external-data", false, "(alpha) Enable external data feature")
+	ExternalDataClientAuthEnabled = flag.Bool("enable-external-data-client-auth", false, "(alpha) Inject Gatekeeper's TLS certificate into an HTTP client that is used to communicate with providers")
```

Comment:
why do we need a feature flag for this?

Reply:
Per https://github.com/open-policy-agent/frameworks/pull/226#discussion_r881077213, reading certificates from disk might cause additional latency so we should let the user opt-in.

Reply:
We're caching the read and watching now, right? That should help things.

Are users able to configure mTLS from the ED resource? If so, does disabling it there disable reading of the cert?

I do like the idea of making it configurable, but definitely don't want to make anything brittle by making it toggle-able in too many places.

Reply:
What's the solution if, for some reason, there is no cert?

Reply:
Is it possible to not have a certificate for the webhook?

Reply:
Yep, you can disable certificate generation and not register a cert on the VWH config/ MWH config. Not that it's recommended, but it's possible

Reply:
Actually, that might not work... it looks like the controller-runtime server would still try to read the certs